client/controller/shared.js

```diff
@@ -109,7 +109,7 @@ export function setUpLocale( context, next ) {
 	} else if ( currentUser ) {
 		context.lang = currentUser.localeSlug;
 	} else {
-		context.lang = context.lang || config( 'i18n_default_locale_slug' );
+		context.lang = config( 'i18n_default_locale_slug' );
```

Comment:
Why do we even set `context.lang` if we overwrite it later at `server/render/index.js`:
```js
	if ( context.store ) {
        ...
		context.isRTL = isRTL( context.store.getState() );
		context.lang = getCurrentLocaleSlug( context.store.getState() );
	}
```

Reply:
I also don't think we should check here for current user, this is already checked on boot,
so IMO the only code needs to be here is so:

```js
export function setUpLocale( context, next ) {
   if ( context.params.lang ) {
      context.store.dispatch( setLocale( context.params.lang ) );
   }
   
 	loadSectionCSS( context, next ); // not sure about that
}
```